### PR TITLE
Precheck Capacity before Provisioning Virtual Machines and Virtual Hard Disks

### PR DESCRIPTION
**Pull request by JamesLiang29**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/Azure/go-autorest/autorest v0.11.29
 	github.com/Azure/go-autorest/autorest/date v0.3.0
 	github.com/google/uuid v1.4.0
-	github.com/microsoft/moc v0.18.1
+	github.com/microsoft/moc v0.19.0-shengl-precheck
```
Comment (JamesLiang29) [inline go.mod:10]: Depending on https://github.com/microsoft/moc/pull/266

File: services/compute/virtualmachine/wssd.go [hunk 2]
```diff
@@ -199,6 +199,39 @@ func (c *client) Validate(ctx context.Context, group, name string) error {
...
 
+func (c *client) Precheck(ctx context.Context, group string, vms []*compute.VirtualMachine) (bool, error) {
+	request, err := c.getVirtualMachinePrecheckRequest(group, vms)
+	if err != nil {
+		return false, err
+	}
+	response, err := c.VirtualMachineAgentClient.Precheck(ctx, request)
+	if err != nil {
+		return false, err
+	} else {
```
Comment (madhanrm) [inline services/compute/virtualmachine/wssd.go:210]: minor: if error return.

code can continue without else. Would keep it cleaner without indentation

File: services/storage/virtualharddisk/wssd.go [hunk 3]
```diff
@@ -79,6 +80,38 @@ func (c *client) Delete(ctx context.Context, group, container, name string) erro
...
+	request, err := getVirtualHardDiskPrecheckRequest(group, container, vhds)
+	if err != nil {
+		return false, err
+	}
+
+	response, err := c.VirtualHardDiskAgentClient.Precheck(ctx, request)
+	if err != nil {
+		return false, err
+	} else {
+		result := response.Result.GetValue()
```
Comment (raghavendra-nataraj) [inline services/storage/virtualharddisk/wssd.go:93]: Can we refactor this result code to a seperate functtion getVirtualHardDiskPrecheckRespose

File: services/compute/virtualmachine/wssd.go (hunk 2 above)
Comment (JamesLiang29) [reply services/compute/virtualmachine/wssd.go]: OK.

File: services/storage/virtualharddisk/wssd.go (hunk 3 above)
Comment (JamesLiang29) [reply services/storage/virtualharddisk/wssd.go]: OK. Also refactor the one for VM.

**Review by raghavendra-nataraj — APPROVED** [review]

**Pull request merged**